server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -75,6 +84,8 @@ public static class Names {
         public static final String SEARCH_COORDINATION = "search_coordination";
         public static final String AUTO_COMPLETE = "auto_complete";
         public static final String SEARCH_THROTTLED = "search_throttled";
+        // Important cluster management tasks. Tasks that manage data, and tasks that report on the cluster health via statistics etc.
+        // It is important that these tasks do not stall because that would block cluster diagnostics.
```

Comment:
Again, not quite sure what you mean by "stalled" here. This pool is used for some IO I think. The point of this pool is for heavy-ish but "background" operations, like stats, where we want to avoid lots of concurrent requests burning all available CPU. If we're under duress, we'd rather stats operations timed out rather than consuming all available resources and affecting "foreground" ops like indexing and search.